engine/src/star_system_xml.cpp

```diff
@@ -1490,6 +1462,63 @@ void StarSystem::LoadXML( const char *filename, const Vector &centroid, const fl
     string file = VSFileSystem::GetCorrectStarSysPath( filename, autogenerated );
     if ( file.empty() )
         file = filename;
+
+    if (game_options.game_speed_affects_autogen_systems)
+        autogenerated = false;
+
+    xml = new Star_XML;
+    xml->scale = 1;
+    xml->fade  = autogenerated;
+    xml->scale *= game_options.star_system_scale;
+    xml->parentterrain = NULL;
+    xml->ct     = NULL;
+    xml->systemcentroid = centroid;
+    xml->timeofyear     = timeofyear;
+    xml->starsp = GetStarSpreadScale();
+    xml->numnearstars   = GetNumNearStarsScale();
+    xml->numstars = GetNumStarsScale();
+    xml->backgroundname = string( "cube" );
+    xml->backgroundColor = GFXColor(1.0,1.0,1.0,1.0);
+    xml->backgroundDegamma = false;
+    xml->reflectivity   = game_options.reflectivity;
+    xml->unitlevel = 0;
+
+
+    VSFile other_file;
+    string full_path = other_file.GetSystemDirectoryPath(file);
+    SystemFactory sys = SystemFactory(file, full_path, xml);
+
+    for(auto& unit : xml->moons)
+    {
+        if (unit->isUnit() == PLANETPTR) {
+            Unit *un = nullptr;
+            // This code here is completely unclear to me and should be refactored
+            // TODO: remove the whole PlanetIterator thing and all custom iterators
+            for (Planet::PlanetIterator iter( (Planet*) unit ); (un = *iter); iter.advance() )
+                AddUnit( un );
+        } else {
+            AddUnit( unit );
+        }
+    }
+
+    createBackground( xml );
+    delete xml;
+}
+
+
+/*void StarSystem::LoadXML( const char *filename, const Vector &centroid, const float timeofyear )
```

Comment:
if we don't need it, we can remove it

Reply:
I've deleted most of the file. Deleting all of it is out of scope of this PR though.